cpp/include/cudf/io/parquet.hpp

```diff
@@ -552,24 +568,22 @@ class parquet_writer_options {
   /**
    * @brief Sets the maximum row group size, in bytes.
    */
-  void set_row_group_size_bytes(size_t size_bytes)
-  {
-    CUDF_EXPECTS(
-      size_bytes >= 512 * 1024,
-      "The maximum row group size cannot be smaller than the page size, which is 512KB.");
-    _row_group_size_bytes = size_bytes;
-  }
+  void set_row_group_size_bytes(size_t size_bytes) { _row_group_size_bytes = size_bytes; }
```

Comment:
Should this validate the maximum row group size against the maximum page size?

I see the comment below suggesting to test parameters on initialization, but that isn't safe if this can be altered after initialization. Perhaps the best approach is to add validation to this method, and make the "builder" interface always sets the page size before the row group size. Then the error can be raised during the builder call OR after initialization.
```suggestion
  void set_row_group_size_bytes(size_t size_bytes)
  {
    CUDF_EXPECTS(
      size_bytes >= _target_page_size_bytes,
      "The maximum row group size in bytes cannot be smaller than the maximum page size in bytes.");
    _row_group_size_bytes = size_bytes;
  }
```

Reply:
yes, I was definitely wanting some help on this logic.  my only concern is with confusion when modifying both row group and page sizes; the page size would have to be changed first.

Reply:
To me it makes the most sense to validate in `build()`. Allows any order and does not delay the validation too much (e.g. like validating in the reader would).

Reply:
@vuule Are you concerned about users calling `set_target_page_size_bytes` after the object is built? Is that allowed by the builder interface?

Reply:
You're right, that won't always work. 
We do have an equivalent interface in ORC. There, this is solved by only having a hard coded minimum for these values instead of comparing the two in the setters. We ensure that the size of the "row group" (called stripe in ORC) is not smaller than the size of "page" (called row group in ORC) in the getters instead of setters! Setters allow _stripes_ to be set to smaller size, but then getters for the _row group_ size return the minimum of the two. Effectively, setting stripe size low also sets the row group size to the same value.

Reply:
@bdice @vuule Should I take a stab at reproducing the ORC logic?

Reply:
@etseidl Yes, that approach seems fine.